invoices/sql_migration_test.go

```diff
@@ -0,0 +1,434 @@
+package invoices
+
+import (
+	"context"
+	crand "crypto/rand"
+	"database/sql"
+	"math/rand"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/lightningnetwork/lnd/channeldb/models"
+	"github.com/lightningnetwork/lnd/clock"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/record"
+	"github.com/lightningnetwork/lnd/sqldb"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	// testHtlcIDSequence is a global counter for generating unique HTLC
+	// IDs.
+	testHtlcIDSequence uint64
+)
+
+// generateTestInvoice generates a random invoice for testing purposes. The
+// invoice will have a random state and random HTLCs. If mpp is true, the
+// invoice will have the MPP feature bit set. If amp is true, the invoice will
+// have the AMP feature bit set and contain AMP HTLCs. Note that this function
+// is not supposed to generate correct invoices, but rather random invoices for
+// testing the SQL migration.
+func generateTestInvoice(t *testing.T, mpp bool, amp bool) *Invoice {
+	var preimage lntypes.Preimage
+	_, err := crand.Read(preimage[:])
+	require.NoError(t, err)
+
+	terms := ContractTerm{
+		FinalCltvDelta:  rand.Int31(),
+		Expiry:          time.Duration(rand.Intn(4444)+1) * time.Minute,
```

Comment:
I believe using constants in this case would actually reduce readability. Since we already have descriptive struct member names, there’s no need to look up additional constants. The code remains clear and easy to understand at a glance.

Reply:
agree